### ISSUE TITLE
Macaroons/Base64 error when trying to log in

### ISSUE THREAD

**Issue by bartaz** [canonical-ols/javan-rhino#101]
Node throws an error (and stops server) when I try to log in:

```
[1] Warning: React.createElement(...): Expected props argument to be a plain object. Properties defined in its prototype chain will be ignored.
[1] /Users/bartaz/Canonical/javan-rhino/node_modules/macaroons.js/lib/Base64Tools.js:20
[1]         return base64.replace(/-/g, '+').replace(/_/g, '/') + Base64Tools.BASE64_PADDING.substr(0, 3 - (base64.length % 3));
[1]                      ^
[1] 
[1] TypeError: Cannot read property 'replace' of undefined
[1]     at Function.Base64Tools.transformBase64UrlSafe2Base64 (/Users/bartaz/Canonical/javan-rhino/node_modules/macaroons.js/lib/Base64Tools.js:20:22)
[1]     at Function.MacaroonsDeSerializer.deserialize (/Users/bartaz/Canonical/javan-rhino/node_modules/macaroons.js/lib/MacaroonsDeSerializer.js:25:43)
[1]     at Function.MacaroonsBuilder.deserialize (/Users/bartaz/Canonical/javan-rhino/node_modules/macaroons.js/lib/MacaroonsBuilder.js:61:38)
[1]     at extractCaveatId (/Users/bartaz/Canonical/javan-rhino/server/macaroons/index.js:12:30)
[1]     at Request._callback (/Users/bartaz/Canonical/javan-rhino/server/handlers/login.js:29:23)
[1]     at Request.self.callback (/Users/bartaz/Canonical/javan-rhino/node_modules/request/request.js:187:22)
[1]     at emitTwo (events.js:106:13)
[1]     at Request.emit (events.js:191:7)
[1]     at Request.<anonymous> (/Users/bartaz/Canonical/javan-rhino/node_modules/request/request.js:1044:10)
[1]     at emitOne (events.js:96:13)
```

**Comment by earnubs** [comment]
https://login.ubuntu.com/ is 503, everything is hosed situation, should be handled with more grace.

**Comment by earnubs** [comment]
fixed in #111
